x/ecocredit/abci.go

```diff
@@ -0,0 +1,18 @@
+package ecocredit
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// BeginBlocker checks if there are any expired sell or buy orders and removes them from state.
+func BeginBlocker(ctx sdk.Context, k Keeper, req abci.RequestBeginBlock) {
+	defer telemetry.ModuleMeasureSince(ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	if err := k.PruneOrders(ctx); err != nil {
+		panic(err)
```

Comment:
I was following the implementation in sdk modules where we panic within `BeginBlocker`.

Reply:
if possible we should avoid panic and use errors whenever possible.

Reply:
I can move the panic to `BeginBlock` but this would go against all other implementations of `BeginBlocker` at the moment, where we panic within `BeginBlocker` and never panic within `BeginBlock`.

Reply:
We have not traditionally returned an error within `BeginBlocker` and panicked within `BeginBlock`. This would be breaking the current pattern but I can make the change if you feel it's necessary. :ok_hand: